### PR TITLE
[CCI] Replace jquery usage in console plugin with native methods

### LINKED ISSUE #3679: Replace `jquery` usage in `console` plugin with native methods
- [x] Understand why and how `jquery` is used in the plugin (add as comments to this issue)
- [ ] Replace `$()` select functions with native methods and properties
- [ ] Replace `$.*` utility methods (for example, `$.each()`, `$.map()`, `$.Deferred()`, ` $.when()`, `$.inArray()`, `$.ajax()`)
- [ ] Remove imports

Prioritize runtime code first, then tests

## Imported in:

```
src/plugins/console/public/application/models/legacy_core_editor/legacy_core_editor.ts:
  30  
  31  import ace from 'brace';
  32  import { Editor as IAceEditor, IEditSession as IAceEditSession } from 'brace';
  33: import $ from 'jquery';
  34  import {
  35    CoreEditor,
  36    Position,

src/plugins/console/public/application/models/legacy_core_editor/__tests__/input.test.js:
  31  import '../legacy_core_editor.test.mocks';
  32  import RowParser from '../../../../lib/row_parser';
  33  import { createTokenIterator } from '../../../factories';
  34: import $ from 'jquery';
  35  import { create } from '../create';
  36  
  37  describe('Input', () => {

src/plugins/console/public/application/models/legacy_core_editor/__tests__/output_tokenization.test.js:
  29   */
  30  
  31  import '../legacy_core_editor.test.mocks';
  32: import $ from 'jquery';
  33  import RowParser from '../../../../lib/row_parser';
  34  import ace from 'brace';
  35  import { createReadOnlyAceEditor } from '../create_readonly';

src/plugins/console/public/application/models/sense_editor/sense_editor.test.mocks.ts:
  32  
  33  import '../legacy_core_editor/legacy_core_editor.test.mocks';
  34  
  35: import jQuery from 'jquery';
  36  jest.spyOn(jQuery, 'ajax').mockImplementation(
  37    () =>
  38      new Promise(() => {

src/plugins/console/public/application/models/sense_editor/__tests__/integration.test.js:
  31  import '../sense_editor.test.mocks';
  32  import { create } from '../create';
  33  import _ from 'lodash';
  34: import $ from 'jquery';
  35  
  36  import * as osd from '../../../../lib/osd/osd';
  37  import * as mappings from '../../../../lib/mappings/mappings';

src/plugins/console/public/application/models/sense_editor/__tests__/sense_editor.test.js:
  30  
  31  import '../sense_editor.test.mocks';
  32  
  33: import $ from 'jquery';
  34  import _ from 'lodash';
  35  
  36  import { create } from '../create';

src/plugins/console/public/lib/ace_token_provider/token_provider.test.ts:
  30  
  31  import '../../application/models/sense_editor/sense_editor.test.mocks';
  32  
  33: import $ from 'jquery';
  34  
  35  // TODO:
  36  // We import from application models as a convenient way to bootstrap loading up of an editor using

src/plugins/console/public/lib/mappings/mappings.js:
  28   * under the License.
  29   */
  30  
  31: import $ from 'jquery';
  32  import _ from 'lodash';
  33  import * as opensearch from '../opensearch/opensearch';
  34  

src/plugins/console/public/lib/osd/osd.js:
  38    UsernameAutocompleteComponent,
  39  } from '../autocomplete/components';
  40  
  41: import $ from 'jquery';
  42  import _ from 'lodash';
  43  
  44  import Api from './api';
```

### PR DESCRIPTION
**Pull request by Nicksqain**
### Description
- [x] Replace `$()` select functions with native methods and properties
- [x] Replace `$.*` utility methods (for example, `$.each()`, `$.map()`, `$.Deferred(`),  `$.when()`, `$.inArray()`, `$.ajax()`)
- [x] Remove imports

### Issues Resolved
- closes #3679
 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by joshuarrrr** [comment]
@Nicksqain Check the failing unit tests - it looks like we may need to account for cases when `mappings` isn't iterable: https://github.com/opensearch-project/OpenSearch-Dashboards/actions/runs/4561381846/jobs/8061888415?pr=3733

**Comment by Nicksqain** [comment]
@joshuarrrr 
изображение
When using fetch and everything else jquery - all tests pass, then this error comes out. How do I run this 127.0.0.1:80?
And if I use everything natively that I've done - then iteration problems. Need to rewrite to loops, most likely

**Comment by Nicksqain** [comment]
@joshuarrrr Can you check pls :]

**Comment by ashwin-pc** [comment]
@Nicksqain As @joshuarrrr pointed out, some of the unit tests seem to be failing. I did not look at each failure, but the last few seem to be because you are trying to map an object without first converting it to an array. `$.each` automatically does this but to reproduce the same behaviour with native js you need to do the conversion first like `Object.entries(obj).forEach(...`

e.g.

```js
$.each(mappings, function (index, indexMapping) {
```

to 

```js
Object.entries(mappings).forEach((index, indexMapping) => {
```

**Comment by joshuarrrr** [comment]
@Nicksqain Instead of running `yarn test:jest`, try with `yarn test:jest_integration` instead, for integration test failures.

**Comment by Nicksqain** [comment]
@joshuarrrr Any guidance on mapping.js? of #3775

**Comment by ashwin-pc** [comment]
@Nicksqain thats annoying merge conflict. Thankfully it looks like you can just accept the current changes in that file since @zhongnansu already removed jquery from that file in his PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -171,6 +171,7 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - [Console] Replace jQuery.ajax with core.http when calling OSD APIs in console ([#3080](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3080))
 - [I18n] Fix Listr type errors and error handlers ([#3629](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3629))
 - [Multiple DataSource] Present the authentication type choices in a drop-down ([#3693](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3693))
+- [Console] Replace jQuery usage in console plugin with native methods([#3733](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3733))
```
Comment (joshuarrrr) [inline CHANGELOG.md:174]:
```suggestion
- [Console] Replace jQuery usage in console plugin with native methods ([#3733](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3733))
```

File: src/plugins/console/public/lib/osd/osd.js [hunk 2]
```diff
@@ -174,20 +173,19 @@ function loadApisFromJson(
...
+      })
+      .then(function (data) {
         setActiveApi(loadApisFromJson(data));
-      },
-      function (jqXHR) {
-        console.log("failed to load API '" + api + "': " + jqXHR.responseText);
-      }
-    );
+      })
+      .catch((error) => console.log("failed to load API '" + api + "': " + error));
```
Comment (joshuarrrr) [inline src/plugins/console/public/lib/osd/osd.js:188]: style nit - prefer template literals to string concatenation
```suggestion
      .catch((error) => console.log(`failed to load API '${api}': ${error}`));
```

**Review by joshuarrrr — DISMISSED** [review]
A couple minor nits.

**Comment by joshuarrrr** [comment]
All checks pass

**Comment by Nicksqain** [comment]
@joshuarrrr  I fixed the error with DCO, but why does it require DCO to merge? Can you fix this? [51da17d](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3733/commits/51da17db37c4b6f9780758e477f96ee235bec04d)

**Comment by joshuarrrr** [comment]
@Nicksqain DCO is part of the project's legal requirements: https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#developer-certificate-of-origin

In this case, the checker is overly strict, in that it expects even commits you accept from reviewers to be signed. Given the content of https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3733/commits/3c7233d3d3b32b6092590b7bd7beaba7f82500e0, I can set the DCO to pass in this case.

**Review by joshuarrrr — APPROVED** [review]

**Review by AMoo-Miki — APPROVED** [review]

**Pull request merged**